src/DynamoCore/Core/DynamoMigrator.cs

```diff
@@ -160,6 +170,18 @@ protected virtual DynamoMigratorBase MigrateFrom(DynamoMigratorBase sourceMigrat
             Copy(sourceMigrator.DefinitionsDirectory, this.DefinitionsDirectory);
 
             this.PreferenceSettings = sourceMigrator.PreferenceSettings;
+            if (this.PreferenceSettings == null) return this;
+
+            // All preference settings are copied over including custom package folders
+            // However if one of the custom folder locations points to the user data directory
+            // of the previous version, it needs to be replaced with that of the current version
+            var indexToreplace = this.PreferenceSettings.CustomPackageFolders.FindIndex(f => f.Contains(sourceMigrator.UserDataDirectory));
+            
+            if (indexToreplace <= -1) return this;
+            
+            this.PreferenceSettings.CustomPackageFolders.RemoveAt(indexToreplace);
+            this.PreferenceSettings.CustomPackageFolders.Insert(indexToreplace, this.UserDataDirectory);
+            
```

Comment:
Can we do `var folders = PreferenceSettings.CustomPackageFolders` earlier since its been used that many times? I'm allergy to long lines of codes.